pkg/cliplugins/bind/plugin/bind_compute.go

```diff
@@ -0,0 +1,302 @@
+	if err != nil {
+		return fmt.Errorf("failed to create kcp client: %w", err)
+	}
+
+	// build config to connect to location workspace
+	kcpConfig := rest.CopyConfig(config)
+	url, _, err := helpers.ParseClusterURL(config.Host)
+	if err != nil {
+		return err
+	}
+
+	kcpConfig.Host = url.String()
+	kcpClient, err := kcpclient.NewClusterForConfig(kcpConfig)
+	if err != nil {
+		return fmt.Errorf("failed to create kcp client: %w", err)
+	}
+
+	supportedExports, err := o.supportedAPIExports(ctx, kcpClient.Cluster(o.LocationWorkspace))
+	if err != nil {
+		return err
+	}
+
+	if err := o.applyAPIBinding(ctx, userWorkspaceKcpClient, supportedExports); err != nil {
+		return err
+	}
+
+	if err := o.applyPlacement(ctx, userWorkspaceKcpClient); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+const maxBindingNamePrefixLength = validation.DNS1123SubdomainMaxLength - 1 - 8
+
+func apiBindingName(clusterName logicalcluster.Name, apiExportName string) string {
+	maxLen := len(apiExportName)
+	if maxLen > maxBindingNamePrefixLength {
+		maxLen = maxBindingNamePrefixLength
+	}
+	bindingNamePrefix := apiExportName[:maxLen]
+
+	hash := sha256.Sum224([]byte(clusterName.Path()))
+	base36hash := strings.ToLower(base36.EncodeBytes(hash[:]))
+	return fmt.Sprintf("%s-%s", bindingNamePrefix, base36hash[:8])
+}
+
+func (o *BindComputeOptions) applyAPIBinding(ctx context.Context, client kcpclient.Interface, desiredAPIExports sets.String) error {
+	apiBindings, err := client.ApisV1alpha1().APIBindings().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	existingAPIExports := sets.NewString()
+	for _, binding := range apiBindings.Items {
+		if binding.Spec.Reference.Workspace == nil {
+			continue
+		}
+		existingAPIExports.Insert(fmt.Sprintf("%s:%s", binding.Spec.Reference.Workspace.Path, binding.Spec.Reference.Workspace.ExportName))
+	}
+
+	diff := desiredAPIExports.Difference(existingAPIExports)
+	var errs []error
+	for export := range diff {
+		clusterName, name := logicalcluster.New(export).Split()
+		apiBinding := &apisv1alpha1.APIBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: apiBindingName(clusterName, name),
+			},
+			Spec: apisv1alpha1.APIBindingSpec{
+				Reference: apisv1alpha1.ExportReference{
+					Workspace: &apisv1alpha1.WorkspaceExportReference{
+						Path:       clusterName.String(),
+						ExportName: name,
+					},
+				},
+			},
+		}
+		_, err := client.ApisV1alpha1().APIBindings().Create(ctx, apiBinding, metav1.CreateOptions{})
+		if err != nil && !errors.IsAlreadyExists(err) {
+			errs = append(errs, err)
+		}
+
+		_, err = fmt.Fprintf(o.Out, "apibinding %s for apiexport %s created.\n", apiBinding.Name, export)
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	return utilerrors.NewAggregate(errs)
+}
+
+// placement name is a hash of location selectors and ns selector, with location workspace name as the prefix
+func (o *BindComputeOptions) placementName() string {
```

Comment:
Why don't we set the default placement name in the Complete() method of the Options ?